pyorbital/tlefile.py

```diff
@@ -343,41 +336,60 @@ def _get_first_tle(uris, open_func, platform=""):
 
 def _get_tles_from_uris(uris, open_func, platform="", only_first=True):
     tles = []
-    designator = "1 " + SATELLITES.get(platform, "")
     for url in uris:
-        fid = open_func(url)
-        for l_0 in fid:
-            tle = ""
-            l_0 = _decode(l_0)
-            if l_0.strip() == platform:
-                l_1 = _decode(next(fid))
-                l_2 = _decode(next(fid))
-                tle = l_1.strip() + "\n" + l_2.strip()
-            elif (platform in SATELLITES or not only_first) and l_0.strip().startswith(designator):
-                l_1 = l_0
-                l_2 = _decode(next(fid))
-                tle = l_1.strip() + "\n" + l_2.strip()
-                if platform:
-                    LOGGER.debug("Found platform %s, ID: %s", platform, SATELLITES[platform])
-            elif open_func == _dummy_open_stringio and l_0.startswith(designator):
-                l_1 = l_0
-                l_2 = _decode(next(fid))
-                tle = l_1.strip() + "\n" + l_2.strip()
-            if tle:
-                if only_first:
-                    return tle
-                tles.append(tle)
+        tles += _get_tles_from_url(url, open_func, platform, only_first)
     if only_first:
+        if tles:
+            return tles[0]
         return ""
     return tles
 
 
+def _get_tles_from_url(url, open_func, platform, only_first):
+    fid = open_func(url)
+    open_is_dummy = open_func == _dummy_open_stringio
+    tles = []
+    for l_0 in fid:
+        tle = _decode_lines(fid, l_0, platform, only_first, open_is_dummy=open_is_dummy)
+        if tle:
+            if only_first:
+                return [tle]
+            tles.append(tle)
+    return tles
+
+
 def _decode(itm):
     if isinstance(itm, str):
         return itm
     return itm.decode("utf-8")
 
 
+def _decode_lines(fid, l_0, platform, only_first, open_is_dummy=False):
+    designator = "1 " + SATELLITES.get(platform, "")
+    tle = ""
+    l_0 = _decode(l_0)
+    if l_0.strip() == platform:
+        l_1 = _decode(next(fid))
+        l_2 = _decode(next(fid))
+        tle = _merge_tle_from_two_lines(l_1, l_2)
+    elif l_0.strip().startswith(designator):
+        if (platform in SATELLITES or not only_first) or open_is_dummy:
+            l_1 = l_0
+            l_2 = _decode(next(fid))
+            tle = _merge_tle_from_two_lines(l_1, l_2)
+            if platform:
+                LOGGER.debug("Found platform %s, ID: %s", platform, SATELLITES[platform])
+    elif l_0.startswith(platform) and platform not in SATELLITES:
+        LOGGER.debug("Found a possible match: %s?", str(l_0.strip()))
```

Comment:
I'm wondering should there be something other than just the log message here? The tests pass, but :man_shrugging:

Reply:
Nope, just wanted to solve the pre-commit complaint and afterwards got confused about the former title and description of the issue :sweat_smile: